src/main/java/cheryl/util/Parser.java

```diff
@@ -0,0 +1,53 @@
+package cheryl.util;
+
+import cheryl.command.AddDeadlineCommand;
+import cheryl.command.AddEventCommand;
+import cheryl.command.AddTodoCommand;
+import cheryl.command.Command;
+import cheryl.command.DeleteCommand;
+import cheryl.command.ExitCommand;
+import cheryl.command.ListCommand;
+import cheryl.command.MarkCommand;
+import cheryl.command.UnmarkCommand;
+
+/**
+ * Parses user input strings into Command objects.
+ * Responsible for determining which command the user wants to execute.
+ */
+
+public class Parser {
+
+    /**
+     * Parses the full command string into a specific Command object.
+     *
+     * @param fullCommand The full user input string
+     * @return A Command object corresponding to the user input
+     * @throws DukeException If the command is unknown or arguments are invalid
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] parts = fullCommand.split(" ", 2);
+        String commandWord = parts[0];
+        String arguments = parts.length > 1 ? parts[1] : "";
+
+        switch (commandWord) {
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+            case "mark":
+                return new MarkCommand(arguments);
+            case "unmark":
+                return new UnmarkCommand(arguments);
+            case "todo":
+                return new AddTodoCommand(arguments);
+            case "deadline":
+                return new AddDeadlineCommand(arguments);
+            case "event":
+                return new AddEventCommand(arguments);
+            case "delete":
+                return new DeleteCommand(arguments);
+            default:
+                throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```

Comment:
Should there be an indentation for switch/case?
Maybe you could consider Lambda-style switch statements/expressions which can have indented case blocks
```suggestion
        return switch (commandWord) {
            case "bye" -> new ExitCommand();
            case "list" -> new ListCommand();
            case "mark" -> new MarkCommand(arguments);
            case "unmark" -> new UnmarkCommand(arguments);
            case "todo" -> new AddTodoCommand(arguments);
            case "deadline" -> new AddDeadlineCommand(arguments);
            case "event" -> new AddEventCommand(arguments);
            case "delete" -> new DeleteCommand(arguments);
            default -> throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
            }
```